OVERVIEW.md

```diff
@@ -0,0 +1,180 @@
+
+## High-Level: What does it do and why do I need it?
+
+> TODO
+
+## Concepts
+
+* Andrew: Reuse and update "Introductions" section from [top level README](readme.md)
+
+
+### Built-in Rule Types
+
+The CloudDQ framework defines 5 basic types of rules:
+
+* NOT_NULL
+* NOT_BLANK
+* REGEX
+* CUSTOM_SQL_EXPRESSION
+* CUSTOM_SQL_STATEMENT
+
+The following table describes the details of these rules and how to use them.
+
+#### Rule Type `NOT_NULL`
+
+Violated if any row in the given column contains null.
+
+The rule is applied to a column in a table (an “entity”)  through a rule binding.
+
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `NOT_BLANK`
+
+Violated if any row in the given column contains null.
+
+The rule is applied to a column in a table (an “entity”)  through a rule binding.
+ 
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `REGEX`
+
+For `STRING` columns. This rule type allows the specification of a Python regular expression to define allowed patterns.
+
+The example shows a check for a valid email address.
+
+The rule is applied to a column in a table through a rule binding.
+
+```yaml
+rules:
+  VALID_EMAIL:
+    rule_type: REGEX
+    dimension: conformance
+    params:
+      pattern: |-
+        ^[^@]+[@]{1}[^@]+$
+
+rule_bindings:
+  VALID_EMAIL:
+    entity_id: TEST_DATA
+    column_id: EMAIL
+    row_filter_id: NONE
+    rule_ids:
+      - VALID_EMAIL
+```
+
+> Complete example TODO
+
+#### Rule Type `CUSTOM_SQL_EXPRESSION`
+
+This rule type allows the specification of a condition using SQL (as in a WHERE clause). The validation fails when the condition is violated.
+
+The example condition performs a SELECT on a reference table to compare to a list of known currencies.
+
+The SQL condition references a parameter $column, which is set through the rule binding.
+
+```yaml
+rules:
+  CORRECT_CURRENCY_CODE:
+    rule_type: CUSTOM_SQL_EXPR
+    dimension: conformance
+    params:
+      custom_sql_expr: |-
+        $column in (select distinct currency_code from `<reference_dataset-id>.currency_codes`)
+
+rule_bindings:
+  LOCATION_LAT:
+    entity_id: TEST_DATA
+    column_id: CURRENCY
+    row_filter_id: NONE
+    rule_ids:
+      - CORRECT_CURRENCY_CODE
+```
+
+> Complete example TODO
+
+
+#### Rule Type `CUSTOM_SQL_STATEMENT`
+
+The “custom SQL statement” allows you to specify a complete SQL statement. Any rows selected by the statement are considered violations of the rule.
+
+The SQL statement references the source data as “data”. The dataset is defined through the entity_id field in the rule binding.
```

Comment:
this is optional - C_S_STMT may or may not reference "data" (and select directly from the underlying table) depending on a particular rule needs. We may want to explain when to exercise each of the scenarios

Reply:
We can change the code but the current code requires the C_S_STMT to contain the string "from data". Are you saying we should change the code?

Reply:
I can confirm that this is not optional as of v0.5.0

Reply:
Andrew to take a note to change this in the future. Closing conversation.